### DISCUSSION TITLE
Trying Docker...I think.

### DISCUSSION THREAD

**Discussion by ahbanks** [lichess-org/fishnet#243] [General]
I'm new to Docker and wanted to give it a spin with fishnet.  Currently, I have a 3 node swarm.  I've tried to create a fishnet service (using docker service create) across these, but I keep getting "Detected task failure" messages and it doesn't look like the service ever starts.  

...any pointers on where I'm going wrong?

**Comment by FirstClassCitizenFCC** [comment]
what command to you run? do you get log output if you run it in foreground mode (without -d switch)?

**Reply by FirstClassCitizenFCC** [reply]
The "detected task failure" message indicates your container fails to start - like you already assumed.
I never used docker swarm. What you can do is try to run it on a single node with the command from the [install-doc](https://github.com/lichess-org/fishnet/blob/master/doc/install.md#docker)
```
docker run -it --name fishnet -e KEY=abcdef niklasf/fishnet:2
```
It shoud print you some log ouput indicating if it is working or not.

You can stop the execution with `Ctrl+C`.

Please provide the log output if your container fails to startup properly.

**Reply by ahbanks** [reply]
The command from the install-doc does work (although it keeps the terminal open, part of what I'm wanting to avoid).

I've tried several variations of `docker service create --name fishnet --replicas=3 niklasf/fishnet -e KEY=<mykey>`

They all grind a while and eventually fail to start.

**Reply by FirstClassCitizenFCC** [reply]
On a single node you can run it in the background if you use the `-d` switch. 
```
docker run -it -d --name fishnet -e KEY=abcdef niklasf/fishnet:2
```
